### PR TITLE
Html5lib tests: Fix reset of context element

### PR DESCRIPTION
**Pull request by sirreal**
Trac ticket: https://core.trac.wordpress.org/ticket/61102

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/html-api/wpHtmlProcessorHtml5lib.php [hunk 1]
```diff
@@ -155,12 +156,14 @@ private static function should_skip_test( $test_name, $expected_tree ): bool {
 	/**
 	 * Generates the tree-like structure represented in the Html5lib tests.
 	 *
-	 * @param string $fragment_context Context element in which to parse HTML, such as BODY or SVG.
+	 * @param string? $fragment_context Context element in which to parse HTML, such as BODY or SVG.
 	 * @param string $html             Given test HTML.
```
Comment (costdev) [inline tests/phpunit/tests/html-api/wpHtmlProcessorHtml5lib.php]:
```suggestion
	 * @param string|null $fragment_context Context element in which to parse HTML, such as BODY or SVG.
	 * @param string      $html             Given test HTML.
```

**Review by costdev — CHANGES_REQUESTED** [review]

**Review by costdev — APPROVED** [review]
Thanks @sirreal, that's all from me on this. I'll leave the rest to @dmsnell as someone with more familiarity of the context (pun intended) :slightly_smiling_face:

**Review by dmsnell — APPROVED** [review]
I'm guessing this was skipping tests because it didn't try the ones that were in `<body>` context? but the others are not in `<body>`?

Looks fine. Thanks for your help, both of you.

**Comment by dmsnell** [comment]
ah. there's more of a description in the Trac ticket. feel free to duplicate descriptions 😉

**Comment by dmsnell** [comment]
Merged in [[58072]](https://core.trac.wordpress.org/changeset/58072)
https://github.com/wordpress/wordpress-develop/commit/a8d5879867e03d62e3a91569aa18c84142e47c62

**Pull request closed without merging**